### PR TITLE
Module restructuring

### PR DESCRIPTION
**Pull request by grisha87**
Introducing changes that align high-level, low-level and experimental modules according to what we agreed to during today's call.

### CODE REVIEW

File: src/golem-network.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { DataTransferProtocol, DeploymentOptions, GolemDeploymentBuilder } from "./deployment";
+import { DataTransferProtocol, DeploymentOptions, GolemDeploymentBuilder } from "./experimental/deployment";
```
Comment (SewerynKras) [inline src/golem-network.ts:1]: If deployment is moved to experimental then there should be a comment on `GolemNetwork.creteDeploymentBuilder` that it's experimental, or that method should be removed altogether

File: src/experimental.ts
Comment (SewerynKras) [inline src/experimental.ts:1]: Experimental modules are now exported in two places, in `src/experimental.ts` and in `src/experimental/index.ts`. 
We should either:
-  delete `src/experimental/index.ts` or
-  replace this file with `export * from './experimental'` or
-  delete this file and change rollup config to use `src/experimental/index.ts`

File: src/golem-network.ts (hunk 1 above)
Comment (SewerynKras) [reply src/golem-network.ts:1]: Pull changes from beta before doing this because after moving this file to `src/golem-network/` git thinks it's a new file

**Review by SewerynKras — CHANGES_REQUESTED** [review]

File: src/experimental.ts
Comment (grisha87) [reply src/experimental.ts:1]: I'm dropping `src/experimental.ts` and adjusting rollup configs.

**Review by SewerynKras — APPROVED** [review]

**Pull request merged**